src/terminal/parser/OutputStateMachineEngine.cpp

```diff
@@ -1408,99 +1409,185 @@ static constexpr bool _isHexNumber(const wchar_t wch) noexcept
+    bool foundRGBColorSpec = false;
     bool foundValidColorSpec = false;
+
+    bool isSharpSignFormat = false;
+    size_t rgbHexDigitCount = 0;
     std::array<unsigned int, 3> colorValues = { 0 };
+    std::array<unsigned int, 3> parameterValues = { 0 };
     bool success = false;
-    // We can have anywhere between [11,15] characters
-    // 9 "rgb:h/h/h"
-    // 12 "rgb:hh/hh/hh"
-    // Any fewer cannot be valid, and any more will be too many.
-    // Return early in this case.
-    //      We'll still have to bounds check when parsing the hh/hh/hh values
-    if (string.size() < 9 || string.size() > 12)
+    const auto stringSize = string.size();
+
+    // First we look for "rgb:"
+    // Other colorspaces are theoretically possible, but we don't support them.
+    auto curr = string.cbegin();
+    if (stringSize > 4)
     {
-        return false;
+        auto prefix = std::wstring(string.substr(0, 4));
+        std::transform(prefix.begin(), prefix.end(), prefix.begin(), ::towlower);
+        if (prefix.compare(L"rgb:") == 0)
+        {
+            // We can have one of the following formats:
+            // 9 "rgb:h/h/h"
+            // 12 "rgb:hh/hh/hh"
+            // 15 "rgb:hhh/hhh/hhh"
+            // 18 "rgb:hhhh/hhhh/hhhh"
+            // Any fewer cannot be valid, and any more will be too many.
+            // Return early in this case.
+            if (stringSize < 9 || stringSize > 18)
+            {
+                return false;
+            }
+
+            foundRGBColorSpec = true;
+
+            std::advance(curr, 4);
+        }
     }
 
-    // Now we look for "rgb:"
-    // Other colorspaces are theoretically possible, but we don't support them.
+    // Try the sharp sign format.
+    if (!foundRGBColorSpec && stringSize > 1)
+    {
+        const auto prefix = string.substr(0, 1);
+        if (prefix.compare(L"#") == 0)
+        {
+            // We can have one of the following formats:
+            // 4 "#hhh"
+            // 7 "#hhhhhh"
+            // 10 "#hhhhhhhhh"
+            // 13 "#hhhhhhhhhhhh"
+            // Any other cases will be invalid.
+            // Return early in this case.
+            if (!(stringSize == 4 || stringSize == 7 || stringSize == 10 || stringSize == 13))
+            {
+                return false;
+            }
 
-    auto curr = string.cbegin();
-    if ((*curr++ == L'r') &&
-        (*curr++ == L'g') &&
-        (*curr++ == L'b') &&
-        (*curr++ == L':'))
+            isSharpSignFormat = true;
+            foundRGBColorSpec = true;
+            rgbHexDigitCount = (stringSize - 1) / 3;
+
+            std::advance(curr, 1);
+        }
+    }
+
+    // Try the XOrg app color name.
+    if (!foundRGBColorSpec)
     {
-        foundRGB = true;
+        try
+        {
+            til::color color = Utils::ColorFromXOrgAppColorName(string);
+            rgb = RGB(color.r, color.g, color.b);
+            success = true;
+        }
+        catch (...)
+        {
+        }

```

Comment:
Is this try/catch actually necessary? Can't we just leave it to be handled by the `CATCH_LOG_RETURN_FALSE` at the end of the method?